Gem/Code/Source/Components/UI/UiMatchPlayerCoinCountsComponent.cpp

```diff
@@ -26,12 +26,15 @@ namespace MultiplayerSample
 
     void UiMatchPlayerCoinCountsComponent::Activate()
     {
+        m_waitForActiveNetworkMatchComponent.Enqueue(AZ::TimeMs{ 1000 }, true);
         StartingPointInput::InputEventNotificationBus::MultiHandler::BusConnect(ShowPlayerCoinCountsEventId);
     }
 
     void UiMatchPlayerCoinCountsComponent::Deactivate()
     {
         StartingPointInput::InputEventNotificationBus::MultiHandler::BusDisconnect();
+        m_waitForActiveNetworkMatchComponent.RemoveFromQueue();
+        m_roundTimerHandler.Disconnect();
```

Comment:
Does this need to detach from m_roundTimerHandler()? Or can the code the base class deactivate instead?

Reply:
This score screen doesn't listen for round changes, so it doesn't have a m_roundNumberHandler.
The reason all these UI screens have a m_waitForActiveNetworkMatchComponent is because they live on a separate entity so cant use the built in component dependency system. Alternatively we could implement a Network Match Component notification when it's activated, and listen for it.